addons/better-emojis/userscript.js

```diff
@@ -1,95 +1,60 @@
-export default async function ({ addon, global, console }) {
-  const updateEmojis = () => {
-    if (addon.tab.clientVersion === "scratch-www") {
-      if (addon.tab.redux?.state?.scratchGui && !addon.tab.redux.state.scratchGui.mode.isPlayerOnly) {
-        return;
-      }
-      const emojis = document.getElementsByClassName("emoji");
-      for (const emoji of emojis) {
-        checkEmojiNew(emoji);
+export default async function ({ addon, global, console, msg }) {
+  const EMOJIS = [
+    "aww-cat",
+    "cat",
+    "cool-cat",
+    "fav-it-cat",
+    "gobo",
+    "huh-cat",
+    "lol-cat",
+    "love-it-cat",
+    "meow",
+    "pizza-cat",
+    "rainbow-cat",
+    "tongue-out-cat",
+    "upside-down-cat",
+    "waffle",
+    "wink-cat",
+  ];
+  const WWW_COMMENTS_REGEX = /^https?:\/\/scratch.mit.edu\/images\/emoji\/(.*).png$/;
+  const R2_COMMENTS_REGEX = /^https:\/\/cdn.scratch.mit.edu\/scratchr2\/static\/(.*)\/images\/easter_eggs\/(.*).png$/;
+  const COMMENTS_REGEX = addon.tab.clientVersion === "scratch-www" ? WWW_COMMENTS_REGEX : R2_COMMENTS_REGEX;
+
+  addon.self.addEventListener("disabled", () => {
+    const emojis = document.querySelectorAll("img.easter-egg, img.emoji");
+    for (const emoji of emojis) {
+      if (EMOJIS.includes(emoji.dataset.emoji)) {
+        // check if emoji is supported by this addon
+        emoji.src =
+          addon.tab.clientVersion === "scratch-www"
+            ? `https://scratch.mit.edu/images/emoji/${emoji.dataset.emoji}.png`
+            : `https://cdn.scratch.mit.edu/scratchr2/static/images/easter_eggs/${emoji.dataset.emoji}.png`;
       }
-    } else if (addon.tab.clientVersion === "scratchr2") {
-      const emojis = document.getElementsByClassName("easter-egg");
-      for (const emoji of emojis) {
-        checkEmoji(emoji);
+    }
+  });
+  addon.self.addEventListener("reenabled", () => {
+    const emojis = document.querySelectorAll("img.easter-egg[data-emoji], img.emoji[data-emoji]");
+    for (const emoji of emojis) {
+      if (EMOJIS.includes(emoji.dataset.emoji)) {
+        // check if emoji is supported by this addon
+        emoji.src = addon.self.dir + "/images/" + emoji.dataset.emoji + ".svg";
       }
     }
-  };
-
-  updateEmojis();
-  const obs = new MutationObserver(updateEmojis);
-  obs.observe(document.body, {
-    childList: true,
-    subtree: true,
   });
+  while (true) {
+    const emoji = await addon.tab.waitForElement(".emoji, .easter-egg", {
+      markAsSeen: true,
+      reduxCondition: (state) => state?.comments.status.comments === "FETCHED",
```

Comment:
`state` is guaranteed to be an object.

Reply:
I think you meant the opposite?
`state.comments?.status?.comments`
Current code in your branch crashes on the messages page